### PR TITLE
Issue147 movers record

### LINKED ISSUE #147: Revise Movers package so that the models take data records.
This integrates https://github.com/lbl-srg/modelica-buildings/issues/209 which was implemented by @Mathadon and myself.

It allows data records to be used for the performance data.

### PR DESCRIPTION
**Pull request by mwetter**
Merge new pumps and fans.
This will close #147

### CODE REVIEW

**Comment by mwetter** [comment]
@damienpicard : Can you please review. This is a big change that we want to have soon on the `master` as I have another major merge to do prepare soon that will affect the `master` and this branch.

**Comment by damienpicard** [comment]
I will look at this tomorrow.

File: Annex60/Fluid/Movers/BaseClasses/Characteristics.mo [hunk 1]
```diff
@@ -3,11 +3,18 @@ package Characteristics "Functions for fan or pump characteristics"
...
+
     parameter Modelica.SIunits.VolumeFlowRate V_flow[:](each min=0)
       "Volume flow rate at user-selected operating points";
-    parameter Modelica.SIunits.Pressure dp[size(V_flow,1)](
+    parameter Modelica.SIunits.Pressure dp[:](
        each min=0, each displayUnit="Pa")
       "Fan or pump total pressure at these flow rates";
+
+    // This is needed for OpenModelica.
+    // fixme: Check if this can be put into FlowMachineInterface instead of here.
```
Comment (damienpicard) [inline Annex60/Fluid/Movers/BaseClasses/Characteristics.mo]: Do you want to keep some *fixme* in the master?

File: Annex60/Fluid/Movers/BaseClasses/ControlledFlowMachine.mo [hunk 2]
```diff
@@ -31,12 +45,11 @@ protected
...
-  V_flow_max=m_flow_nominal/rho_default;
+
 equation
   r_V = VMachine_flow/V_flow_max;
-  etaHyd = cha.efficiency(data=hydraulicEfficiency, r_V=r_V, d=hydDer);
-  etaMot = cha.efficiency(data=motorEfficiency,     r_V=r_V, d=motDer);
+  etaHyd = cha.efficiency(per=per.hydraulicEfficiency, V_flow=VMachine_flow, d=hydDer, r_N=1, delta=1E-4);
+  etaMot = cha.efficiency(per=per.motorEfficiency,     V_flow=VMachine_flow, d=motDer, r_N=1, delta=1E-4);
   dpMachine = -dp;
   VMachine_flow = -port_b.m_flow/rho_in;
```
Comment (damienpicard) [inline Annex60/Fluid/Movers/BaseClasses/ControlledFlowMachine.mo]: Why is `VMachine_flow = - port_b.m_flow/rho_in` and not `=port_a.m_flow/rho_in`? `rho_in` is the density of inlet fluid...

File: Annex60/Fluid/Movers/BaseClasses/FlowControlledMachine.mo [hunk 3]
```diff
@@ -1,6 +1,10 @@
 within Annex60.Fluid.Movers.BaseClasses;
-partial model PrescribedFlowMachine
+partial model FlowControlledMachine
```
Comment (damienpicard) [inline Annex60/Fluid/Movers/BaseClasses/FlowControlledMachine.mo]: Confusing names! `ControlledFlowMachine` and `FlowControlledMachine` are very similar names. Why not using the same names as the actual pumps: `FlowControlledMachine` and 'SpeedControlledMachine` instead?

File: Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo [hunk 4]
```diff
@@ -36,92 +30,157 @@ partial model FlowMachineInterface
...
+                zeros(nOri + 1)
+             elseif haveVMax then
+              cat(1, {dpMax}, {_per_y.pressure.dp[i] + _per_y.pressure.V_flow[i] * kRes for i in 1:nOri})
+             elseif haveDPMax then
+              cat(1, {_per_y.pressure.dp[i] + _per_y.pressure.V_flow[i] * kRes for i in 1:nOri}, {0})
+             else
+               zeros(nOri+1))
     "Volume flow rate vs. total pressure rise with correction for pump resistance added";
+                                  // fixme check whether these branches are correct
+                                  // fixme check whether these branches are correct
```
Comment (damienpicard) [inline Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo]: Do these `fixme` still need to be solved?

**Comment by damienpicard** [comment]
For me, this branch can be merged after having addressed my few comments.

File: Annex60/Fluid/Movers/BaseClasses/ControlledFlowMachine.mo (hunk 2 above)
Comment (mwetter) [inline Annex60/Fluid/Movers/BaseClasses/ControlledFlowMachine.mo]: I will correct this. Using `port_a.m_flow` is better.

File: Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo (hunk 4 above)
Comment (mwetter) [inline Annex60/Fluid/Movers/BaseClasses/FlowMachineInterface.mo]: They are no longer needed. I will remove them.

File: Annex60/Fluid/Movers/BaseClasses/FlowControlledMachine.mo (hunk 3 above)
Comment (mwetter) [inline Annex60/Fluid/Movers/BaseClasses/FlowControlledMachine.mo]: I agree.  Will rename them as
```
Annex60/Fluid/Movers/BaseClasses/ControlledFlowMachine.mo -> Annex60/Fluid/Movers/BaseClasses/FlowControlled.mo
Annex60/Fluid/Movers/BaseClasses/FlowControlledMachine.mo -> Annex60/Fluid/Movers/BaseClasses/SpeedControlled.mo
```

**Pull request merged**